app/Models/Entity.php

```diff
@@ -438,4 +438,19 @@ public function postPositionOptions($position = null): array
         }*/
         return $options;
     }
+
+    /**
+     * Type filter link
+     */
+    public function typeLink(string $displayName = null): string
+    {
+        $campaign = CampaignLocalization::getCampaign();
+        $defaultOptions = auth()->check() && auth()->user()->entityExplore === '1' ? [$campaign, 'm' => 'table'] : [$campaign];
+
+        return '<a href="' .
```

Comment:
I try to avoid writing html in models. You can use `return link_to_route()` instead, and let laravel figure out how to write the html :)